utils/frame/remote-externalities/src/lib.rs

```diff
@@ -323,16 +327,23 @@ where
 	B::Hash: DeserializeOwned,
 	B::Header: DeserializeOwned,
 {
+	const MAX_PARALLELISM: usize = 4;
 	const BATCH_SIZE_INCREASE_FACTOR: f32 = 1.10;
 	const BATCH_SIZE_DECREASE_FACTOR: f32 = 0.50;
 	const INITIAL_BATCH_SIZE: usize = 5000;
 	// NOTE: increasing this value does not seem to impact speed all that much.
 	const DEFAULT_KEY_DOWNLOAD_PAGE: u32 = 1000;
 
 	/// Get the number of threads to use.
+	/// Cap the number of threads. Performance improvement beyond a small number of threads is
+	/// negligible, and too many threads can create issues with the HttpClient.
 	fn threads() -> NonZeroUsize {
-		thread::available_parallelism()
-			.unwrap_or(NonZeroUsize::new(1usize).expect("4 is non-zero; qed"))
+		let avaliable = thread::available_parallelism()
+			.unwrap_or(NonZeroUsize::new(1usize).expect("1 is non-zero; qed"))
+			.get();
+		assert!(avaliable > 0, "avaliable parallelism must be greater than 0");
```

Comment:
You could also add an env var check like `TRY_RUNTIME_MAX_THREADS`, that would be easier than to add a CLI argument, but a bit hacky.

Reply:
Sure, added a `TRY_RUNTIME_MAX_THREADS` env var